examples/example_1_simple_monoblock.py

```diff
@@ -1,37 +1,52 @@
-from sledo.optimiser import Optimiser
-from sledo.simulation import ThermoMechSimulation
-
-DATA_DIR = "./results/example_1"
-
-search_space = (
-    {
-        "name": "Armour height",
-        "type": "range",
-        "hit_name": "monoBArmHeight",
-        "hit_block": "",
-        "bounds": [1e-3, 20e-3],
-    },
-    {
-        "name": "Block width",
-        "type": "range",
-        "hit_name": "monoBWidth",
-        "hit_block": "",
-        "bounds": [17e-3, 34e-3],
-    },
-)
-
-opt = Optimiser(
-    name="simple_monoblock",
-    search_space=search_space,
-    simulation_class=ThermoMechSimulation,
-    data_dir=DATA_DIR,
-)
-
-opt.load_input_file("./input_files/simple_monoblock.i")
-opt.run_optimisation_loop(
-    objective_name="stress",
-    max_iter=20,
-    minimise=True,
-)
-
-opt.pickle()
+"""
+SLEDO Simple Monoblock Example
+
+Example input file for running Bayesian optimisation of a simplified divertor
+monoblock to minimise the peak von misses stress predicted by a
+thermomechanical MOOSE simulation.
+
+(c) Copyright UKAEA 2023-2024.
+"""
+
+from ray import tune
+
+from sledo import Optimiser, MooseHerderDesignEvaluator
+from sledo import SLEDO_ROOT, MOOSE_CONFIG_FILE
+
+EXAMPLES_DIR = SLEDO_ROOT / "examples"
+INPUT_FILE = EXAMPLES_DIR / "input_files" / "simple_monoblock_thermomech.i"
+WORKING_DIR = EXAMPLES_DIR / "results" / "example_1"
+METRICS = ["max_stress"]
+
+if __name__ == "__main__":
+
+    # Instantiate design evaluator.
+    design_evaluator = MooseHerderDesignEvaluator(
+        METRICS,
+        INPUT_FILE,
+        working_dir=WORKING_DIR,
+        config_path=MOOSE_CONFIG_FILE,
+    )
+
+    # Define a search space according to the Ray Tune API.
+    # Documentation here:
+    # https://docs.ray.io/en/latest/tune/api/search_space.html
+    search_space = {
+        "monoBArmHeight": tune.uniform(1e-3, 20e-3),
+        "monoBThick": tune.uniform(0.5e-3, 9e-3),
+    }
+
+    # Instantiate SLEDO optimiser.
+    opt = Optimiser(
+        "simple-monoblock-optimiser",
```

Comment:
`Optimiser` doesn't know about the `.i` and indeed no `.i` exists at this stage for the catbird and test function `DesignEvaluator`s. I would like to use a sensible default if possible though. The only design-specific strings passed to `Optimiser` are those in `metrics`, could use a concatenation of metrics for the name if one isn't passed?

Reply:
I've done this for now such that we get something like `"{metric1}_{metric2}_{etc.}_optimiser"` if no name is passed, so for the monoblock it would name the optimiser `max_stress_optimiser`.